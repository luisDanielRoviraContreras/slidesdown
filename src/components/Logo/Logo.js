import React from 'react'
import PropTypes from 'prop-types'

import { StyledImg } from './Logo.style'

import logoSvg from './logo.svg'

const Logo = ({ onClick, large }) => (
  <StyledImg
    alt="Slidesdown"
    large={large}
    onClick={onClick || null}
    src={logoSvg}
  />
)

Logo.propTypes = {
  large: PropTypes.bool,
  onClick: PropTypes.func
}

export default Logo
