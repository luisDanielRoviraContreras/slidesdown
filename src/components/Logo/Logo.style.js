import styled from 'styled-components'

export const StyledImg = styled.img`
  height: 2rem;
  width: 2rem;
`
