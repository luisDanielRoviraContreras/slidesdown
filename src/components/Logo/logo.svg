<?xml version="1.0" encoding="UTF-8"?>
<svg width="40px" height="40px" viewBox="0 0 40 40" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 48.2 (47327) - http://www.bohemiancoding.com/sketch -->
    <title>Logo Copy 2</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <linearGradient x1="0%" y1="0%" x2="101.999998%" y2="100.999999%" id="linearGradient-1">
            <stop stop-color="#3023AE" offset="0%"></stop>
            <stop stop-color="#C86DD7" offset="100%"></stop>
        </linearGradient>
        <rect id="path-2" x="0" y="0" width="32" height="24"></rect>
        <rect id="path-3" x="0" y="0" width="32" height="24"></rect>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Logo-Copy-2">
            <g id="Group" transform="translate(1.000000, 3.000000)">
                <g id="Slide-Copy" transform="translate(17.000000, 14.000000) rotate(-5.000000) translate(-17.000000, -14.000000) translate(1.000000, 2.000000)">
                    <g id="Combined-Shape">
                        <use fill="#FFFFFF" fill-rule="evenodd" xlink:href="#path-2"></use>
                        <rect stroke="url(#linearGradient-1)" stroke-width="1" x="0.5" y="0.5" width="31" height="23"></rect>
                    </g>
                    <path d="M18.41518,12.64 L20.81518,12.64 L16.01518,16.64 L11.21518,12.64 L13.61518,12.64 L13.61518,7.52 L18.41518,7.52 L18.41518,12.64 Z" id="Combined-Shape" fill="url(#linearGradient-1)" fill-rule="evenodd"></path>
                </g>
                <g id="Slide-Copy" transform="translate(21.000000, 21.000000) rotate(5.000000) translate(-21.000000, -21.000000) translate(5.000000, 9.000000)">
                    <g id="Combined-Shape">
                        <use fill="#FFFFFF" fill-rule="evenodd" xlink:href="#path-3"></use>
                        <rect stroke="url(#linearGradient-1)" stroke-width="1" x="0.5" y="0.5" width="31" height="23"></rect>
                    </g>
                    <path d="M18.41518,12.64 L20.81518,12.64 L16.01518,16.64 L11.21518,12.64 L13.61518,12.64 L13.61518,7.52 L18.41518,7.52 L18.41518,12.64 Z" id="Combined-Shape" fill="url(#linearGradient-1)" fill-rule="evenodd"></path>
                </g>
            </g>
        </g>
    </g>
</svg>