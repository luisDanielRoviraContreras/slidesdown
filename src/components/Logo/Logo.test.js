import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Logo from './Logo'

jest.mock('./logo.svg', () => 'Logo');

it('renders correctly', () => {
  const tree = renderer.create(<Logo />).toJSON()

  expect(tree).toMatchSnapshot()
})
