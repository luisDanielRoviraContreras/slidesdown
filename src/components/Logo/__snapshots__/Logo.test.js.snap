// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  height: 40px;
  margin-right: 0.5rem;
  width: 40px;
}

<div
  className="c0"
>
  <img
    className="c1"
    src="Logo"
  />
   Slidesdown
</div>
`;
